addons/web/static/src/legacy/js/core/smooth_scroll_on_drag.js

```diff
@@ -391,7 +391,15 @@ const SmoothScrollOnDrag = Class.extend(mixins.ParentedMixin, {
      */
     _onSmoothDragStart(ev, ui, onDragStartCallBack) {
         this.scrollTargetIsDocument = this.$scrollTarget.is('html');
-        this.iframeOffset = false;
+        const iframeEl = this.$scrollTarget[0].ownerDocument.defaultView.frameElement;
+        const draggableInsideIframe = iframeEl
+            && iframeEl.contentDocument.contains(this.$element[0]);
+        // If the scrolling element is within an iframe and the draggable
+        // element is outside this iframe, the offsets must be computed taking
+        // into account the iframe.
+        this.iframeOffset = draggableInsideIframe
+                            ? false
+                            : iframeEl && iframeEl.getBoundingClientRect();
```

Comment:
I would rewrite the code like this to make it easier to understand (it will prevent rechecking if `iframeEl` exists too) :
```js
this.iframeOffset = false;
// If the scrolling element is within an iframe and the draggable
// element is outside this iframe, the offsets must be computed taking
// into account the iframe.
const iframeEl = this.$scrollTarget[0].ownerDocument.defaultView.frameElement;
if (iframeEl && !iframeEl.contentDocument.contains(this.$element[0])) {
  this.iframeOffset = iframeEl.getBoundingClientRect();
}
```
(just a suggestion :wink:)

Reply:
> ```js
> this.iframeOffset = false;
> // If the scrolling element is within an iframe and the draggable
> // element is outside this iframe, the offsets must be computed taking
> // into account the iframe.
> const iframeEl = this.$scrollTarget[0].ownerDocument.defaultView.frameElement;
> if (iframeEl && !iframeEl.contentDocument.contains(this.$element[0])) {
>   this.iframeOffset = iframeEl.getBoundingClientRect();
> }
> ```

really better, thanks